addons/l10n_in_ewaybill_stock/models/l10n_in_ewaybill.py

```diff
@@ -0,0 +1,630 @@
+
+    content = fields.Binary(compute='_compute_content', compute_sudo=True)
+    cancel_reason = fields.Selection(selection=[
+        ("1", "Duplicate"),
+        ("2", "Data Entry Mistake"),
+        ("3", "Order Cancelled"),
+        ("4", "Others"),
+    ], string="Cancel reason", copy=False, tracking=True)
+    cancel_remarks = fields.Char("Cancel remarks", copy=False, tracking=True)
+
+    def _compute_supply_type(self):
+        for ewaybill in self:
+            ewaybill.supply_type = ewaybill.picking_type_code == 'incoming' and 'I' or 'O'
+
+    @api.depends('picking_id')
+    def _compute_document_partners_details(self):
+        for ewaybill in self.filtered(lambda ewb: ewb.state == 'pending'):
+            picking_id = ewaybill.picking_id
+            if ewaybill.picking_type_code == 'incoming':
+                ewaybill.partner_bill_to_id = picking_id.company_id.partner_id
+                ewaybill.partner_bill_from_id = picking_id.partner_id
+                ewaybill.partner_ship_to_id = picking_id.picking_type_id.warehouse_id.partner_id
+                ewaybill.partner_ship_from_id = picking_id.partner_id
+            else:
+                ewaybill.partner_bill_to_id = picking_id.partner_id
+                ewaybill.partner_bill_from_id = picking_id.company_id.partner_id
+                ewaybill.partner_ship_to_id = picking_id.partner_id
+                ewaybill.partner_ship_from_id = picking_id.picking_type_id.warehouse_id.partner_id
+                if partner_invoice_id := ewaybill.picking_id._l10n_in_get_invoice_partner():
+                    ewaybill.partner_bill_to_id = partner_invoice_id
+
+            if (
+                ewaybill.picking_type_code == 'dropship' and
+                (dest_partner := ewaybill.picking_id._get_l10n_in_dropship_dest_partner())
+            ):
+                ewaybill.partner_ship_to_id = dest_partner
+                ewaybill.partner_ship_from_id = ewaybill.picking_id.partner_id
+
+    @api.depends('fiscal_position_id')
+    def _compute_is_fiscal_position_invisible(self):
+        """
+        Make fiscal field invisible if a Sales Order (SO) or Purchase Order (PO) exists.
+
+        This function checks the presence of a Sales Order (SO) or Purchase Order (PO) in the system.
+        If either a SO or PO exists, the fiscal field will be made invisible.
+        `_l10n_in_get_fiscal_position` will return False in this case.
+
+        However, if no SO or PO exists, the fiscal field will be visible.
+        If a fiscal position is not set on the SO or PO, the field will still be invisible,
+        returning an empty recordset.
+
+        Example:
+        - self.env['account.fiscal.position'] == False (output will be False)
+        - self.env['account.fiscal.position'] == account.fiscal.position() (output will be True)
+        """
+        for ewaybill in self:
+            fiscal_position = ewaybill.picking_id._l10n_in_get_fiscal_position()
```

Comment:
We can make it simpler by letting _l10n_in_get_fiscal_position() (maybe change to _l10n_in_get_fiscal_position_info) return two params: should_be_forced, fiscal_position.